### PR TITLE
Update RELEASE.md to add release lifecycle

### PR DESCRIPTION
**Pull request by sherry-x**
### Description

- describe in high level on how a release lifecycle looks like
- add release announcement method so that people know where to follow the update

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -35,6 +35,22 @@ The aptos framework release tags are created to facilitate the on-chain framewor
 ### aptos-cli-v*X.Y.Z* release tag
 The aptos cli release tags are created to track the CLI versions for community to use when developing on the Aptos network. It's always recommended to upgrade your CLI when a new version is released, for the best user experience. Learn how to update to the [latest CLI version](https://aptos.dev/tools/install-cli/install-from-brew/#upgrading-the-cli).
 
+## Aptos Release Lifecycle
+(The time length here is a rough estimate, it varies depends on each release.)
+* [day 0] A release branch `aptos-release-vx.y` will be created, with a commit hash `abcde`. The full test suite will be triggered for the commit hash for validation.
+* [day 1] The release will be deployed to **devnet**.
+* [day 7] Once the release passed devnet test, a release tag `aptos-release-vx.y.z` will be created, and get deployed to **testnet**.
```
Comment (rustielin) [inline RELEASE.md:42]: correction: release tag is called `aptos-node-vx.y.z`, though the release branch is called `aptos-release-vx.y`. Maybe add a note somewhere that this is for the `aptos-node` release specifically, and that CLI etc have their own release cadences?

File: RELEASE.md [hunk 2]
```diff
@@ -35,6 +35,22 @@ The aptos framework release tags are created to facilitate the on-chain framewor
...
+* [day 14] After the binary release stablized on testnet, testnet framework will be upgraded.
+* Hot-fixes release will be created as needed when a release version is soaking in testnet, and we will only promote a release from testnet to Mainnet after confirming a release version is stable.
+* [day 24] Once confirmed that both binary upgrade and framework upgrade stablized on testnet, the release version will be deployed to 1% of the stake on **Mainnet**.
+* [day 25] Wider announcement will be made for the community to upgrade the binary, `aptos-release-vx.y.z` will be updated with "[Mainnet]" in the release page, Mainnet validators will be slowly upgrading.
```
Comment (rustielin) [inline RELEASE.md:46]: same as above, `aptos-node-vx.y.z`

File: RELEASE.md [hunk 3]
```diff
@@ -35,6 +35,22 @@ The aptos framework release tags are created to facilitate the on-chain framewor
...
+* [day 30] A list of framework upgrade proposals will be submitted to Mainnet for voting.
+* [day 37] Proposals executed on-chain if passed voting.
+
+## Release Announcement
+* Each of the network release will be announced on Aptos Network [Discord](https://discord.gg/aptosnetwork). Follow mainnet-release, testnet-release, devnet-release channel to get updates.
+* When a release is ready to deploy, a Github release page will be created in this repo as well. You can search for the most recent release version titled with "[Mainnet]" for production usage.
```
Comment (rustielin) [inline RELEASE.md:52]: maybe include a link ? https://github.com/aptos-labs/aptos-core/releases

**Review by rustielin — APPROVED** [review]
LGTM % minor corrections/nits

**Review by thepomeranian — APPROVED** [review]

**Review by saharct — APPROVED** [review]

**Pull request merged**